src/shared/sleep-config.c

```diff
@@ -224,13 +221,22 @@ static int hibernation_partition_size(size_t *size, size_t *used) {
                         continue;
                 }
 
-                if (streq(type, "partition") && endswith(dev, "\\040(deleted)")) {
-                        log_warning("Ignoring deleted swapfile '%s'.", dev);
+                if (streq(type_field, "partition") && endswith(dev_field, "\\040(deleted)")) {
+                        log_warning("Ignoring deleted swapfile '%s'.", dev_field);
                         continue;
                 }
-
-                *size = size_field;
-                *used = used_field;
+                if (device) {
+                        *device = dev_field;
+                        dev_field = NULL;
```

Comment:
please use the TAKE_PTR() macro for the above (It's a very recent addition to systemd git, but makes passing of ownership much nicer and shorter)